test/lib/analyzer/image-inspector.test.ts

```diff
@@ -36,3 +39,140 @@ test("image id", async (t) => {
   t.same(imageData.Id, expectedId, "id as expected");
   t.same(imageData.RootFS.Layers, expectedLayers, "layers as expected");
 });
+
+test("extract image details", async (t) => {
+  const tests = {
+    "hello-world": {
+      expected: {
+        hostname: "registry-1.docker.io",
+        imageName: "library/hello-world",
+        tag: "latest",
+      },
+    },
+    "gcr.io/kubernetes/someImage:alpine": {
+      expected: {
+        hostname: "gcr.io",
+        imageName: "kubernetes/someImage",
+        tag: "alpine",
+      },
+    },
+    "nginx:1.18": {
+      expected: {
+        hostname: "registry-1.docker.io",
+        imageName: "library/nginx",
+        tag: "1.18",
+      },
+    },
+    "calico/cni:release-v3.14": {
+      expected: {
+        hostname: "registry-1.docker.io",
+        imageName: "calico/cni",
+        tag: "release-v3.14",
+      },
+    },
```

Comment:
Could we also add a test for the localhost case and for the registry-with-port case?

Reply:
added some, lmk what you think